src/common/snippets/src/runtime_configurator.cpp

```diff
@@ -272,10 +266,187 @@ void RuntimeConfigurator::update_data_offsets() const {
+    std::vector<std::vector<size_t>> layouts(m_io_num);
+    for (size_t i = 0; i < m_io_num; ++i)
+        layouts[i] = m_io_descs[i]->get_layout();
+    return layouts;
+}
+
+RuntimeConfigurator::UnifiedLoopInfoRtParams RuntimeConfigurator::compute_runtime_params(const lowered::UnifiedLoopInfoPtr& loop_info) {
+    RuntimeConfigurator::UnifiedLoopInfoRtParams rt_params;
+    rt_params.work_amount = loop_info->get_work_amount();
+    const auto count = loop_info->get_input_count() + loop_info->get_output_count();
+    rt_params.ptr_increments.resize(count);
+    rt_params.finalization_offsets.resize(count);
+
+    size_t idx = 0;
+    loop_info->iterate_through_descs(
+        [&rt_params, &idx](const lowered::UnifiedLoopInfo::LoopPortDesc& desc) {
+            rt_params.ptr_increments[idx] = desc.ptr_increment;
+            rt_params.finalization_offsets[idx] = desc.finalization_offset;
+            ++idx;
+        });
+    return rt_params;
+}
+
+void RuntimeConfigurator::ParallelWAOptimizer::init(const lowered::LinearIRPtr& linear_ir,
+                                                    const std::vector<PortDescriptorPtr>& io_descs,
+                                                    size_t in_num) {
+    if (linear_ir->get_config().m_enable_domain_optimization || !linear_ir->is_dynamic())
+        return;
+
+    const auto brgemms = find_applicable_brgemms(linear_ir);
+    // Parallel WA optimization is Brgemm related
+    if (brgemms.empty())
+        return;
+
+    concurrency = linear_ir->get_config().m_min_parallel_work_amount;
+    unsqueezed_params = find_unsqueezed_params(linear_ir, brgemms);
+    OPENVINO_ASSERT(!unsqueezed_params.empty(), "unsqueezed_params mustn't be empty after initialization");
+    loops_to_split = find_loops_to_split(linear_ir, unsqueezed_params);
+
+    m_dim_idces.resize(io_descs.size());
+    optimized_layouts.resize(io_descs.size());
+    for (size_t i = 0; i < io_descs.size(); ++i) {
+        const auto& layout = io_descs[i]->get_layout();
+        const auto dim_idx = i < in_num ? utils::get_input_dim_idx(layout, 1)
+                                        : utils::get_output_dim_idx(layout, 1);
+        m_dim_idces[i] = dim_idx;
+        optimized_layouts[i] = SplitDimensionM::get_updated_order(layout, i < in_num ? dim_idx : layout.size() - 2);
+    }
+}
+
+bool RuntimeConfigurator::ParallelWAOptimizer::optimize(VectorDims& master_shape,
+                                                        RuntimeConfigurator::LoopInfoRuntimeParamsMap& map,
+                                                        std::vector<ov::snippets::VectorDims>& shapes,
+                                                        std::vector<std::vector<size_t>>& layouts,
+                                                        size_t in_num) {
+    size_t new_batch_dim, new_kernel_dim;
+    if (loops_to_split.empty() || !SplitDimensionM::split(master_shape, concurrency, new_batch_dim, new_kernel_dim))
```

Comment:
It's so sad that we don't have pass pipeline here))))

I'm thinking that we should use another check instead of `loops_to_split.empty()`.

Probably instead of non obvious check `loops_to_split.empty()` we may add new attribute `enabled` to `RuntimeConfigurator`.
```
// ticket -> this struct should be pass
struct Optimizer {
    bool enabled = true;
    ParallelWAOptimizer func {};
} optimizer;

RuntimeConfigurator::init() {
    optimizer.enabled = optimizer.func.init(linear_ir, m_io_descs, m_in_num);
}

RuntimeConfigurator::update() {
    if (optimizer_enabled)
           m_optimizer.func.optimize();
}
```

What do you think? It's some emulation `ov::pass`. For sure, need to think about naming))
I hope we will rewrite it anyway

Reply:
Let's introduce `enabled` method which will encapsulate the current unobvious logic. If `update`, will be rewritten on PassPipeline, we will just remove the method. I also attached a ticket number to a method's description